pkg/api/controller.go

```diff
@@ -2107,6 +2107,150 @@ func (c *Controller) ResetBranch(w http.ResponseWriter, r *http.Request, body Re
+	}) {
+		return
+	}
+
+	ctx := r.Context()
+	c.LogAction(ctx, "import", r, repository, branch, "")
+	user, err := auth.GetUser(ctx)
+	if err != nil {
+		writeError(w, r, http.StatusUnauthorized, "missing user")
+		return
+	}
+	var metadata map[string]string
+	if body.Commit.Metadata != nil {
+		metadata = body.Commit.Metadata.AdditionalProperties
+	}
+	paths := make([]catalog.ImportPath, 0, len(body.Paths))
+	for _, p := range body.Paths {
+		pathType, err := catalog.GetImportPathType(p.Type)
+		if c.handleAPIError(ctx, w, r, err) {
+			return
+		}
+		paths = append(paths, catalog.ImportPath{
+			Destination: p.Destination,
+			Path:        p.Path,
+			Type:        pathType,
+		})
+	}
+
+	committer := user.Username
+	importID, err := c.Catalog.Import(r.Context(), repository, branch, catalog.ImportRequest{
+		Paths: paths,
+		Commit: catalog.ImportCommit{
+			CommitMessage: body.Commit.Message,
+			Committer:     committer,
+			Metadata:      metadata,
+		},
+	})
+	if c.handleAPIError(ctx, w, r, err) {
+		return
+	}
+	writeResponse(w, r, http.StatusCreated, ImportCreationResponse{
+		Id: importID,
+	})
+}
+
+func importStatusToResponse(status *graveler.ImportStatus) ImportStatusResp {
+	resp := ImportStatusResp{
+		Completed:      status.Completed,
+		ImportBranch:   status.ImportBranch,
+		ImportProgress: status.Progress,
+		UpdateTime:     status.UpdatedAt,
+	}
+
+	if status.Error != nil {
+		resp.Error = &Error{Message: status.Error.Error()}
+	}
+	if status.MetaRangeID != nil {
+		metarange := status.MetaRangeID.String()
+		resp.MetarangeId = &metarange
+	}
+	commitLog := catalog.CommitRecordToLog(status.Commit)
+	if commitLog != nil {
+		resp.Commit = &Commit{
+			Committer:    commitLog.Committer,
+			CreationDate: commitLog.CreationDate.Unix(),
+			Id:           commitLog.Reference,
+			Message:      commitLog.Message,
+			MetaRangeId:  commitLog.MetaRangeID,
+			Metadata:     &Commit_Metadata{AdditionalProperties: commitLog.Metadata},
+			Parents:      commitLog.Parents,
+		}
+	}
+
+	return resp
+}
+
+func (c *Controller) ImportStatus(w http.ResponseWriter, r *http.Request, repository, branch string, params ImportStatusParams) {
+	if !c.authorize(w, r, permissions.Node{
+		Permission: permissions.Permission{
+			Action:   permissions.ReadBranchAction,
+			Resource: permissions.BranchArn(repository, branch),
+		},
+	}) {
+		return
+	}
+	ctx := r.Context()
+	c.LogAction(ctx, "get_import_status", r, repository, branch, "")
+	status, err := c.Catalog.GetImportStatus(ctx, repository, params.Id)
+	if c.handleAPIError(ctx, w, r, err) {
+		return
+	}
+
+	resp := importStatusToResponse(status)
+	writeResponse(w, r, http.StatusOK, resp)
+}
+
+func (c *Controller) ImportCancel(w http.ResponseWriter, r *http.Request, repository, branch string, params ImportCancelParams) {
+	if !c.authorize(w, r, permissions.Node{
+		Permission: permissions.Permission{
+			Action:   permissions.ImportFromStorage,
+			Resource: permissions.StorageNamespace("*"),
+		},
```

Comment:
We only get the import ID, what do you think the permissions should be for canceling an import job?

Reply:
The StorageNamespace in this permission refers to the import source - so that doesn't make sense to allow import cancel only for users who have permission to import from a specific source.